fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkcommon/BulkDataUtils.java

```diff
@@ -157,8 +158,11 @@ public static int readFhirResourceFromObjectStore(AmazonS3 cosClient, String buc
            int numOfLinesToSkip, List<Resource> fhirResources) {
         int exported;
         S3Object item = cosClient.getObject(new GetObjectRequest(bucketName, itemName));
-        try (BufferedReader resReader = new BufferedReader(new InputStreamReader(item.getObjectContent()))) {
+        try (S3ObjectInputStream s3InStream = item.getObjectContent();
+             BufferedReader resReader = new BufferedReader(new InputStreamReader(s3InStream))) {
             exported = getFhirResourceFromBufferReader(resReader, numOfLinesToSkip, fhirResources);
+            // Notify s3 client to abort and prevent the server from keeping on sending data.
+            s3InStream.abort();
```

Comment:
Is abort for exception handling logic? if so, why is it in the main `try`?

Reply:
it's not for exception handling logic, if exception happens, the exported will be 0, and the job will be finished with error message in the log, so there will be no issue.   This is for the normal flow of the chunk job, e.g, in first chunk, the job opens the objectstream, read 1000 lines and then abort the stream, and then writes to fhir server, persistence 1000 to checkpoint as the tobeskipped lines, and then in the following second chunk, the job opens the objectstream again, skip 1000 lines first, then read 1000 lines and then abort the stream, and then writes to fhir server, and persistence 2000 to checkpoint as the tobeskipped lines, repeat the above steps till all the lines are processed.      Note: In current implementation,  I didn't provide option to reuse the inputstream/reader across all the chunk operations, because I thought the to-be-imported file can be very huge, and this can cause the connection to be occupied by the import job for too long time which in theory may cause issue for the server if too many concurrent huge files are imported at the same time.  But I did have opened issue #726 to provide option(e.g, controlled by a switch) to reuse the inputstream/reader across all chunk operations if the job is not paused or stopped and continued.

Reply:
I get it! this makes sense.